src/query_graph/path_tree.rs

```diff
@@ -111,13 +111,59 @@ impl OpPathTree {
         }
         Ok(true)
     }
+
+    fn fmt_internal(
+        &self,
+        f: &mut Formatter<'_>,
+        indent: &String,
+        include_conditions: bool,
+    ) -> std::fmt::Result {
+        if self.is_leaf() {
+            return writeln!(f, "{}", self.vertex());
+        }
+        writeln!(f, "{}:", self.vertex())?;
+        let child_indent = indent.to_owned() + "  ";
+        for child in self.childs.iter() {
+            let index = match child.edge {
+                Some(index) => index,
+                None => EdgeIndex::end(),
+            };
```

Comment:
There's a function for that! 😄 
```suggestion
            let index = child.edge.unwrap_or_else(EdgeIndex::end);
```